controllers/nstemplateset/tier_template_revision.go

```diff
@@ -0,0 +1,34 @@
+package nstemplateset
+
+import (
+	"context"
+	"fmt"
+
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/api/v1alpha1"
+	"github.com/codeready-toolchain/toolchain-common/pkg/cluster"
+
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// getTierTemplateRevision gets the TierTemplateRevision resource from the host cluster.
```

Comment:
Nitpick: the name doesn't match the real function name (and I personally like the name without the "Toolchain" better) :)